scripts/patch-kernel.sh

```diff
@@ -36,6 +36,10 @@ for i in ${patchdir}/${patchpattern} ; do
     esac
     [ -d "${i}" ] && echo "Ignoring subdirectory ${i}" && continue	
     echo ""
+    # because some people publish tarballs containing files with modes u-w
+    awk -e '/^\+\+\+ / { n = match($2, "/"); print substr($2, n + 1); }' "${i}" | while read file; do
+	[ ! -f ${targetdir}/$file -o -w ${targetdir}/$file ] || chmod u+w ${targetdir}/$file
```

Comment:
Actually, I tried your suggestion but when the patch creates new files there's obviously no file to chmod u+w on, so this causes a lot of noise like:

```
$ grep '^chmod:' logs/target/linux/compile.txt
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/net/netfilter/nf_conntrack_rtcache.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/net/netfilter/nf_conntrack_rtcache.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/scripts/mod/elf.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/tools/include/tools/linux_types.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/arch/arm/boot/compressed/piggy.cat.S': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/lib/decompress_uncat.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/arch/mips/lib/memcmp.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/arch/mips/kernel/machine_kexec.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/fs/jffs2/compr_lzma.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/linux/lzma.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/linux/lzma/LzFind.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/linux/lzma/LzHash.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/linux/lzma/LzmaDec.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/linux/lzma/LzmaEnc.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/linux/lzma/Types.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/lib/lzma/LzFind.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/lib/lzma/LzmaDec.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/lib/lzma/LzmaEnc.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/lib/lzma/Makefile': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/drivers/net/phy/mdio-boardinfo.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/drivers/net/phy/mdio-boardinfo.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/linux/platform_data/phy-at803x.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/drivers/rtc/rtc-rtc7301.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/drivers/rtc/rtc-pt7c4338.c': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/include/linux/crashlog.h': No such file or directory
chmod: cannot access '/home/philipp/bertram/lede/build_dir/target-x86_64_westmere_musl_powercode-bmu/linux-x86_xeon/linux-4.9.14/kernel/crashlog.c': No such file or directory
$ 
```

So I'm reverting to the originally proposed fix.